src/services/oneSignal/OneSignalManager.tsx

```diff
@@ -1,21 +1,44 @@
-import React, { useEffect } from 'react'
+import React, { useEffect, useState, useContext } from 'react'
+import { strict as assert } from 'assert'
 import OneSignal from 'react-onesignal'
+import { ProfileContext } from '../../context/profile'
+import { TokenContext } from '../../context/token'
+import { createSubscriptionIdOneSignal, getSubscriptionIdOneSignal } from '../api'
+
+const OneSignalManager: React.FC<any> = ({ children }) => {
+  const { profile } = useContext(ProfileContext)
+  const { token } = useContext(TokenContext)
+  const [oneSignalInitialized, setOneSignalInitialized] = useState<boolean>(false)
 
-const OneSignalManager: React.FC = ({ children }) => {
   useEffect(() => {
+    if (profile?.id === 0 || profile?.id == null || token == null || token.trim() === '') return
+
     const initializeOneSignal = async (): Promise<void> => {
-      try {
+      if (!oneSignalInitialized) {
         await OneSignal.init({
           appId: process.env.REACT_APP_ONE_SIGNAL_ID as string,
-          allowLocalhostAsSecureOrigin: true
+          safari_web_id: process.env.REACT_APP_SAFARI_WEB_ID as string,
+          allowLocalhostAsSecureOrigin: true,
+          notifyButton: { enable: true },
+          size: 'small'
         })
+        console.log('OneSignal initialized successfully')
+        setOneSignalInitialized(true)
+      }
+      const result = await getSubscriptionIdOneSignal(token)
+      if (Array.isArray(result) && result.length > 0) return
```

Comment:
Am I missing something or this signature isn't matching with what I'm currently seeing in `SubscriptionInfo`?